tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -705,9 +706,14 @@ def prepare_solution_vector(self, number_density):
         atomic_numbers = number_density.index
         solution_array = []
         for atomic_number in atomic_numbers:
+            needed_number_of_elements = rate_matrix_index[
+                rate_matrix_index.get_level_values("atomic_number")
+                == atomic_number
```

Comment:
`(rate_matrix_index.get_level_values("atomic_number") == atomic_number).sum()` would be shorter and has one indexing operation less.